tests/unit/fork_test.cpp

```diff
@@ -0,0 +1,52 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <unistd.h>
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+using namespace testing;
+
+extern "C" {
+   #include <tilck/kernel/process.h>
+   #include "mocking.h"
+   int vfs_dup(fs_handle h, fs_handle *dup_h);
+   void vfs_close(fs_handle h);
+   int fork_dup_all_handles(struct process *pi);
+}
+
+class vfs_mock : public KernelSingleton {
+public:
+
+   MOCK_METHOD(int, vfs_dup, (fs_handle h, fs_handle *dup_h), (override));
+   MOCK_METHOD(void, vfs_close, (fs_handle h), (override));
+};
+
+TEST(fork_dup_all_handles, trigger_inside_path) {
+   vfs_mock mock;
+   process pi;
+   bzero(&pi, sizeof(process));
+   fs_handle_base handle0_dup, handle1_dup, handle2_dup;
+
+   pi.handles[0] = (fs_handle) &handle0_dup;
+   pi.handles[1] = (fs_handle) &handle1_dup;
+   pi.handles[2] = (fs_handle) &handle2_dup;
```

Comment:
**BUG:** Actually, that's wrong here. Here we should set the original handles, not the duplicated ones. We need theoretically 3 handles (original) and 2 duplicate ones. You can fake that this way:

```
   pi.handles[0] = TO_PTR(0xaa00);
   pi.handles[1] = TO_PTR(0xaa01);
   pi.handles[2] = TO_PTR(0xaa02);
```

Those pointers are **invalid**, but that's fine because we mocked `vfs_dup()`.

Reply:
Or, as suggested below, if you prefer the test to be more realistic and less "tricky", you can just allocate an array of 3 handles and another array of 2 dup_handles.